### ISSUE TITLE
Unable to see NDI source from streaming laptop

### ISSUE THREAD

**Issue by slavonicsniper** [Palakis/obs-ndi#166]
Here is collection of all my posts regarding my issue:

Hello, I just tried NDI for the first time yesterday. First I tried it with my older laptop that is connected to my home network over the WIFI. Everything was working, my laptop could see my main PC as NDI source and add it. Then I tried the same thing with my newer laptop that is connected to my home network via etherner cable. I did everything the same as with the older laptop but my newer laptop cannot see my my PC desktop as source name in NDI when I want to add it BUT my main PC dekstop can see this laptop so I get like only one way communication. I tried to disable firewall, no success. I installed the same OBS version, same obs-ndi plugin I restarted both computers. I tried to reset my network settings.

Any ideas what could be wrong if I can see my laptop from my main PC where I am gaming but I can't see my main PC from laptop where I'd like to stream? I don't think there is anything wrong with my main PC as I could see it as a NDI source from my older laptop that was connected over WIFI.

Thanks for help guys

I tried both suggestions but still no success. I am running out my ideas and solutions...the thing is that it worked 2 days ago on my main PC, but I had the latest windows update 1803 installed that time, yesterday I downgraded back to the previous version as I had stuttering issues after the latest WIN update. From that time my NDI plugin in OBS stopped working no laptop in my network can see it as a source in OBS. Other laptops can see each other. I tried to do clean reinstallation many times and in different ways but anyways it doesn't work...what else could I try guys? any ideas? I really want to use NDI and the saddest thing is that it was already working and now it stopped somehow...

so to recap: it was already working on my main pc where now this pc cannot be recognized by other pcs in laptop, firewall exception is set correctly everywhere

**Comment by Helgaiden** [comment]
You'll need to add the computer specs, specific things you tried, and OBS logs from each PC as the problems were happening to get any real help man.

**Comment by slavonicsniper** [comment]
Hello everyone,

I am glad to tell you that I've just fixed my issue. The problem was "Microsoft KM-TEST Loopback Adapter" that I created in past for my network labs. Thanks to this [PDF](https://233b1d13b450eb6b33b4-ac2a33202ef9b63045cbb3afca178df8.ssl.cf1.rackcdn.com/pdf/newtek-ndi-problem-solving.pdf) I could know what ports is NDI using for network communication. So there are 5353 and 5960 (read PDF for more info about those ports). Then I checked how OBS with NDI plugin is actually listening on those ports. Why? Other devices with NDI plugin could see each other on my network but couldn't see my main laptop, so I wanted to see and compare if it isn't listening in a different wrong way on my main PC. So here is output from netstat:

C:\Users\erikg>netstat -aon | findstr 10380
  TCP    0.0.0.0:5960           0.0.0.0:0              LISTENING       10380
  TCP    0.0.0.0:5961           0.0.0.0:0              LISTENING       10380
  TCP    0.0.0.0:49479          0.0.0.0:0              LISTENING       10380
  TCP    127.0.0.1:49479        127.0.0.1:51012        ESTABLISHED     10380
  TCP    127.0.0.1:51012        127.0.0.1:49479        ESTABLISHED     10380
  TCP    192.168.1.127:51023    52.15.232.153:443      ESTABLISHED     10380
  TCP    192.168.1.127:51024    54.69.249.94:443       ESTABLISHED     10380
  **UDP    0.0.0.0:5353           *:*                                    10380**
  **UDP    192.168.1.3:5353       *:*                                    10380**
  UDP    192.168.1.3:8005       *:*                                    10380
  UDP    192.168.1.3:10669      *:*                                    10380
  UDP    192.168.1.3:52690      *:*                                    10380
  UDP    192.168.1.3:52691      *:*                                    10380
  **UDP    192.168.1.127:5353     *:*                                    10380**
  UDP    192.168.1.127:8005     *:*                                    10380
  UDP    192.168.1.127:10669    *:*                                    10380
  UDP    192.168.1.127:52692    *:*                                    10380
  UDP    192.168.1.127:52693    *:*                                    10380
  **UDP    192.168.56.1:5353      *:*                                    10380**
  UDP    192.168.56.1:8005      *:*                                    10380
  UDP    192.168.56.1:10669     *:*                                    10380
  UDP    192.168.56.1:52688     *:*                                    10380
  UDP    192.168.56.1:52689     *:*                                    10380

so as you can see there are actually 4 IP's that are listening on this UDP port (5353) that is important so the other devices can see your PC and add it as NDI source on your network:

0.0.0.0
192.168.1.3 -> IP set on my (Microsoft KM-TEST Loopback Adapter)
192.168.1.127 -> IP set by my wifi router
192.168.56.1 -> IP set by (Npcap Loopback Adapter) - it's virtual box adapter host-only

First I thought it's alright because as you can see there is 0.0.0.0 first so it should actually listen on any IP right? But anyways it was working yet. So I tried to disable my "192.168.1.3 -> IP set on my (Microsoft KM-TEST Loopback Adapter)" and then the output of netstat was like this:

C:\Users\erikg>netstat -aon | findstr 4780
  TCP    0.0.0.0:5960           0.0.0.0:0              LISTENING       4780
  TCP    0.0.0.0:5961           0.0.0.0:0              LISTENING       4780
  TCP    0.0.0.0:49479          0.0.0.0:0              LISTENING       4780
  TCP    127.0.0.1:49479        127.0.0.1:51061        ESTABLISHED     4780
  TCP    127.0.0.1:51061        127.0.0.1:49479        ESTABLISHED     4780
  TCP    192.168.1.127:51062    104.20.83.218:443      ESTABLISHED     4780
  TCP    192.168.1.127:51063    216.58.201.78:443      ESTABLISHED     4780
  TCP    192.168.1.127:51064    216.58.201.110:443     ESTABLISHED     4780
  TCP    192.168.1.127:51065    74.125.133.154:443     ESTABLISHED     4780
  TCP    192.168.1.127:51066    216.58.201.98:443      ESTABLISHED     4780
  TCP    192.168.1.127:51067    92.123.17.82:443       ESTABLISHED     4780
  TCP    192.168.1.127:51068    52.15.232.153:443      ESTABLISHED     4780
  TCP    192.168.1.127:51069    92.123.17.82:443       ESTABLISHED     4780
  TCP    192.168.1.127:51070    34.215.68.228:443      ESTABLISHED     4780
  **UDP    0.0.0.0:5353           *:*                                    4780
  UDP    192.168.1.127:5353     *:*                                    4780**
  UDP    192.168.1.127:8005     *:*                                    4780
  UDP    192.168.1.127:10669    *:*                                    4780
  UDP    192.168.1.127:60410    *:*                                    4780
  UDP    192.168.1.127:60411    *:*                                    4780
  **UDP    192.168.56.1:5353      *:*                                    4780**
  UDP    192.168.56.1:8005      *:*                                    4780
  UDP    192.168.56.1:10669     *:*                                    4780
  UDP    192.168.56.1:60408     *:*                                    4780
  UDP    192.168.56.1:60409     *:*                                    4780

so now my main IP '192.168.1.127' set by wifi router that I am using to communicate with other devices on my LAN network at home is first in order after 0.0.0.0 IP address. So let's try to see if my my PC is visible on my streaming laptop. And? There you go it is! :)

So maybe a message to developers, try to check this. Why does the order matter if the port is listening on every possible IP's on the PC? Why did my 'Microsoft KM-TEST Loopback Adapter'  cause a problem if the port is also listening on 0.0.0.0 which should mean any address... I hope this will help. I was troubleshooting it few days btw.

**Comment by Helgaiden** [comment]
@Palakis might want to see this

**Comment by Palakis** [comment]
@slavonicsniper I think the order matters because of mDNS discovery: it picks the first actual IP (not 0.0.0.0 or localhost) available so it can use it to advertise itself on the network.

What happened was: with the loopback enabled, the NDI library picked this IP and used it to advertise NDI on ALL networks the machine is connected to. Problem is, that loopback IP obviously doesn't exist on your main network so no connection was possible.

**Comment by Palakis** [comment]
NewTek announced that NDI 3.5 fixes a lot of discovery and network issues, so it may be good to give this a try again once the new obs-ndi with 3.5 support gets released.

**Comment by slavonicsniper** [comment]
Thanks for info and resolving

**Comment by dpnz** [comment]
I couldn't make sense of the solution from slavonicsniper (although I assume this is what happened to me re the IP stuff) however I managed to get a fix/workaround by uninstalling OBS and the plugin and then reinstalling the 32Bit OBS (instead of 64bit default) and then the plugin and now I can see both sources on both computers.
